src/Axe/Controllers/CompilerApiController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Axe.Managers;
+using Axe.ViewModels.CompilerVm;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Axe.Models;
+using Newtonsoft.Json;
+
+namespace Axe.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/compiler/[action]")]
+    public class CompilerApiController : Controller
+    {
+        private readonly ICompilerManager compilerManager;
+        private readonly UserManager<ApplicationUser> userManager;
+
+        public CompilerApiController(ICompilerManager compilerManager, UserManager<ApplicationUser> userManager)
+        {
+            this.compilerManager = compilerManager;
+            this.userManager = userManager;
+        }
+
+        [HttpGet]
+        public async Task<CodeBlockVm> GetTaskById(int id)
```

Comment:
why this method is allowed to anyone, when other two require [Authorize]? include that clarification in comments for methods whcih currently missing